File: summary/ruby/11_prime_factorization.rb
def prime_factors(n)
  factors = []
  divisor = 2

  while n > 1
    while n % divisor == 0
      factors << divisor
      n /= divisor
    end
    divisor += 1
  end

  factors
end

n = gets.chomp.to_i

puts "素因数分解結果: #{prime_factors(n)}"

File: summary/go/11_prime_factorization.go
package main

import (
	"fmt"
)

func primeFactors(n int) []int {
	var factors []int
	divisor := 2

	for n > 1 {
		for n%divisor == 0 {
			factors = append(factors, divisor)
			n /= divisor
		}
		divisor++
	}

	return factors
}

func main() {
	var n int
	fmt.Print("整数を入力してください: ")
	fmt.Scan(&n)

	fmt.Printf("素因数分解結果: %v\n", primeFactors(n))
}
